doc/adr/0009-support-reference-scripts-in-constraint-library.rst

```diff
@@ -0,0 +1,42 @@
+.. _support_reference_scripts_in_constraint_library:
+
+ADR 9: Support reference scripts in constraint library
+======================================================
+
+Date: 2022-08-15
+
+Author(s)
+---------
+
+koslambrou <konstantinos.lambrou@iohk.io>
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+In Babbage era (available after the Vasil HF), the Cardano blockchain will support "reference scripts" by changing the ``TxOut`` data type.
+Reference scripts are used to attach arbitrary scripts to transaction outputs and are used to satisfy script requirements during validation, rather than requiring the spending transaction to do so.
+Thus, we need to adapt our transaction constraint data type (``TxConstraints``) to support this new feature.
+
+Decision
+--------
+
+* We will add ``Maybe ScriptHash`` as a new data constructor parameter for the constraints ``MustPayToPubKeyAddress``, ``MustPayToOtherScript``, ``ScriptOutputConstraint`` in ``TxConstraints``.
+  In the offchain implementation of those constraints, if a reference script hash is provided, we will need to find actual script in the lookups table so that we can include it in the transaction output.
```

Comment:
Suggest adding "the": 

... we will need to find the actual script ...

Reply:
```suggestion
  In the off-chain implementation of those constraints, if a reference script hash is provided, we will need to find the actual script in the lookups table so that we can include it in the transaction output.
```